.github/workflows/client.yml

```diff
@@ -134,6 +142,37 @@ jobs:
             REVISION=${{ env.revision }}
           push: true
 
+      - name: Build Client Binaries
+        uses: docker/build-push-action@v3
+        with:
+          target: output-bins
+          outputs: type=local,dest=./out/bin/
+          build-args: |
+            ENVIRONMENT=${{ github.event.inputs.environment }}
+            VERSION=${{ env.version }}
+            REVISION=${{ env.revision }}
+          push: false
+
+      - name: Archive Client Binaries
```

Comment:
Just so I would understand - is this step here just so that we would be able to manually download the binaries from the details of the run if we want to check them out?

Reply:
We want to publish binaries when releasing a new version, not just the Docker image. The binaries built in the previous step are compressed and we compute hashes for them. Then, they are published to a bucket. We do not want to build binaries locally, we want the CI to do that for us.

Reply:
Yeah, I get that, but this particular step (using `actions/upload-artifact@v3` action) is not part of the building of binaries or publishing them to a bucket (unless I'm missing something?). It just archives all the files mentioned in the `path` param to one file called `binaries` and makes the file accessible for download in the GH page with the details of the workflow run. The file could be also used to pass it to some other job in the workflow via the `actions/download-artifact` action, but we don't use it like that in this workflow.
So what I wanted to confirm is that this action is used just to create an archive with the binary and the hashes that can be downloaded via GH UI. Or is there some other purpose that I'm missing?

Reply:
> Just so I would understand - is this step here just so that we would be able to manually download the binaries from the details of the run if we want to check them out?

Yes, this is just for us in case we want to verify the CI process. The actual publication happens in the GCP upload step.